### PR TITLE
Update README and models page with Qwen2 models

### PR DESCRIPTION
**Pull request by joecummings**
#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [ ] fix a bug
- [x] update tests and/or documentation
- [ ] other (please add here)

#### Changelog
* README
* Models page (very basic)

#### Test plan

IIIIIIIIIIIIIIIIIIIIIII

(get it, b/c it's a bunch of I's)

### CODE REVIEW

File: docs/source/api_ref_models.rst [hunk 1]
```diff
@@ -133,6 +133,29 @@ To download the CodeLlama-7B model:
     code_llama2.lora_code_llama2_70b
     code_llama2.qlora_code_llama2_70b
 
+qwen-2
+------
+
+Models from the `Qwen2 small model family <https://huggingface.co/collections/Qwen/qwen2-6659360b33528ced941e557f>`_.
```
Comment (SalmanMohammadi) [inline docs/source/api_ref_models.rst:139]:
```suggestion
Models of size 0.5B, 1.5B, and 7B from the `Qwen2 family <https://huggingface.co/collections/Qwen/qwen2-6659360b33528ced941e557f>`_.
```

File: docs/source/api_ref_models.rst [hunk 2]
```diff
@@ -133,6 +133,29 @@ To download the CodeLlama-7B model:
...
+
+To download the Qwen2 1.5B model, for example:
+
+.. code-block:: bash
+
+    tune download Qwen/Qwen2-0.5B-Instruct --output-dir /tmp/Qwen2-0.5B-Instruct --ignore-patterns None
```
Comment (SalmanMohammadi) [inline docs/source/api_ref_models.rst:145]:
```suggestion
    tune download Qwen/Qwen2-1.5B-Instruct --output-dir /tmp/Qwen2-1.5B-Instruct --ignore-patterns None
```

File: docs/source/api_ref_models.rst [hunk 3]
```diff
@@ -133,6 +133,29 @@ To download the CodeLlama-7B model:
...
+
+.. autosummary::
+    :toctree: generated/
+    :nosignatures:
+
```
Comment (SalmanMohammadi) [inline docs/source/api_ref_models.rst:150]: we also usually expose the component builders, but let's chat about what the best way to doc this is in the model page PR, I think we agree this doesn't look very clear to the uninitiated
```suggestion
    qwen2.qwen2
    qwen2.lora_qwen2
```

File: docs/source/api_ref_models.rst [hunk 4]
```diff
@@ -133,6 +133,29 @@ To download the CodeLlama-7B model:
...
+    qwen2.qwen2_7b
+    qwen2.qwen2_0_5b
+    qwen2.qwen2_1_5b
+    qwen2.lora_qwen2_7b
+    qwen2.lora_qwen2_0_5b
+    qwen2.lora_qwen2_1_5b
+    qwen2.qwen2_tokenizer
+    qwen2.Qwen2Tokenizer
```
Comment (SalmanMohammadi) [inline docs/source/api_ref_models.rst:158]: think you need to add this to the init

Comment (SalmanMohammadi) [reply docs/source/api_ref_models.rst:158]: should be a doozy, pal

Comment (SalmanMohammadi) [reply docs/source/api_ref_models.rst:158]: you got this boss

**Comment by SalmanMohammadi** [comment]
nice pr, buddy

File: docs/source/api_ref_models.rst (hunk 4 above)
Comment (joecummings) [reply docs/source/api_ref_models.rst:158]: image

**Comment by joecummings** [comment]
> nice pr, buddy

Thanks pal, I live for the praise

**Review by SalmanMohammadi — APPROVED** [review]

**Pull request merged**